common/src/main/java/com/netflix/conductor/common/metadata/workflow/WorkflowTask.java

```diff
@@ -569,6 +569,13 @@ public WorkflowTask next(String taskReferenceName, WorkflowTask parent) {
 							return nextTask;
 						}
 						if (task.has(taskReferenceName)) {
+							if (TaskType.DO_WHILE.name().equals(task.getType())) {
```

Comment:
@alex-fu Since we are diverging the DECISION block by so many lines of code. We can define a separate switch case for DO_WHILE task. And move common code to one function.

Reply:
OK, I'll move this common code to a function. But this function should be called from DECISION block and FORK block, not DO_WHILE block(since we don't allow nested DO_WHILE task).